textdata_preprocessor.py

```diff
@@ -0,0 +1,282 @@
+# Copyright 2021 The Petuum Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+# pylint: disable=attribute-defined-outside-init
+"""
+IU Xray medical report preprocessor for the text that does extraction
+of findings and impression part of the text. Tokenize the sentence, lowercase
+all the characters and remove the word contains non-alphabetic characters
+"""
+from typing import Iterator, Any
+import os.path as osp
+import argparse
+import xml.etree.ElementTree as ET
+from collections import Counter
+
+from ft.onto.base_ontology import Token, Document
+from forte.data.data_pack import DataPack
+from forte.processors.base import PackProcessor
+from forte.data.readers.base_reader import PackReader
+from forte.pipeline import Pipeline
+from forte.processors.nltk_processors import NLTKWordTokenizer
+from forte.data.multi_pack import MultiPack
+from forte.processors.base import MultiPackProcessor
+from forte.processors.writers import PackNameJsonPackWriter
+from forte.data.caster import MultiPackBoxer
+from forte.data.selector import NameMatchSelector
+from forte.data.data_utils_io import dataset_path_iterator
+
+from iu_xray.onto.iu_xray_ontology import Impression, Findings, \
+    FilePath, ParentImage
+
+
+counter = Counter()
```

Comment:
Let's do not use this global `Counter` shared by multiple classes. 

The pipeline object has a `resources` field that is shared. You can put the `Counter` inside the `resources`:

https://github.com/asyml/forte/blob/master/forte/pipeline.py#L120

i.e: 
Add it like:
`pipeline.resources.update('counter', Counter())`

Use it during all processors:
`self.resources.get('counter')`

Reply:
modified